### PR TITLE
SOLR-15535: remove rawtypes warnings in 'grouping' code

### PR DESCRIPTION
**Pull request by cpoerschke**
https://issues.apache.org/jira/browse/SOLR-15535

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/search/grouping/distributed/shardresultserializer/SearchGroupsResultTransformer.java [hunk 1]
```diff
@@ -85,7 +85,9 @@ public SearchGroupsResultTransformer(SolrIndexSearcher searcher) {
         searchGroup.groupValue = new BytesRef(groupValue);
       }
     }
-    searchGroup.sortValues = rawSearchGroupData.toArray(new Comparable[rawSearchGroupData.size()]);
+    @SuppressWarnings("unchecked")
+    Comparable<?>[] sv = (Comparable<?>[]) Array.newInstance(Comparable.class, rawSearchGroupData.size());
```
Comment (madrob) [inline solr/core/src/java/org/apache/solr/search/grouping/distributed/shardresultserializer/SearchGroupsResultTransformer.java:89]: I think you can do this, but I can't find a JLS reference for why it works.
```suggestion
    Comparable<?>[] sv = new Comparable<?>[rawSearchGroupData.size()];
```

**Comment by madrob** [comment]
I tried to mess around with this a bit by creating a wrapping class, but couldn't get it to work very cleanly so let's stick with this approach and maybe can revisit in the future. Thanks for picking it up!

**Comment by madrob** [comment]
I think this is ok to merge?

**Comment by cpoerschke** [comment]
Yeah, thanks for the ping. I'd seen that it was "approved" with words but not "button" and so I'd let it sit for a bit but yeah time to merge now.

**Pull request merged**